components/brave_rewards/browser/database/database_contribution_queue.cc

```diff
@@ -148,8 +156,13 @@ ledger::ContributionQueuePtr DatabaseContributionQueue::GetFirstRecord(
 bool DatabaseContributionQueue::DeleteRecord(
     sql::Database* db,
     const uint64_t id) {
-  if (!db->Execute("PRAGMA foreign_keys=1;")) {
-    LOG(ERROR) << "Error: deleting record for contribution queue with id" << id;
+  DCHECK(db);
+  if (id == 0 || !db) {
```

Comment:
`NOTREACHED()` ?

Reply:
no need IMHO

Reply:
If we do not add a `NOTREACHED()` and db is null during development we will not pick up these edge case bugs

Reply:
of course we will pick it up as we have DCHECK one line above for db

Reply:
oh yeah :-) I missed that one